src/scratchpads/completion_single_file_fim.rs

```diff
@@ -212,20 +207,55 @@ impl ScratchpadAbstract for SingleFileFIM {
             );
         } else if self.order == "SPM" {
             prompt = format!(
-                "{}{}{}{}{}{}{}{}{}",
+                "{}{}{}{}{}{}{}{}",
                 self.t.eos,
                 self.fim_suffix,
-                extra_context,
                 cursor_line2,
                 after,
                 self.fim_prefix,
-                before.into_iter().rev().collect::<Vec<_>>().join(""),
+                before,
                 cursor_line1,
                 self.fim_middle,
             );
         } else {
             return Err(format!("order \"{}\" not recognized", self.order));
         }
+        let prompt_tokens_before_context = self.t.count_tokens(prompt.as_str())?;
```

Comment:
Do we really need that? Encoding text is a long operation. We don't have a counter above?